components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/cache/IdPResourceIdCacheKey.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.idp.mgt.cache;
+
+import org.wso2.carbon.identity.application.common.cache.CacheKey;
+
+/**
+ * Cache key for IDP Cache by resource ID.
+ */
+public class IdPResourceIdCacheKey extends CacheKey {
+
+    private static final long serialVersionUID = 8614291529494760436L;
+    private String resourceId;
+
+    public IdPResourceIdCacheKey(String resourceId) {
+        this.resourceId = resourceId;
+    }
+
+    public String getResourceId() {
+        return getResourceId();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
+            return false;
+        }
+
+        IdPResourceIdCacheKey that = (IdPResourceIdCacheKey) o;
+
+        if (!resourceId.equals(that.resourceId)) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = super.hashCode();
+        result = 31 * result + resourceId.hashCode();
```

Comment:
what is represented by 31? shall we have a constant for this?

Reply:
in Java, hashcode for string function uses 31 as a multiplier
https://coderanch.com/t/458720/java/Java-hashCode-String-multiplier

Reply:
In this case, value of super.hashCode is not required since 'resourceId' is unique accross tenant. Hence, i have updated.